relay-dynamic-config/src/global.rs

```diff
@@ -117,6 +129,24 @@ fn is_default<T: Default + PartialEq>(t: &T) -> bool {
     *t == T::default()
 }
 
+fn default_on_error<'de, D, T>(deserializer: D) -> Result<T, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+    T: Default + serde::de::DeserializeOwned,
+{
+    match T::deserialize(deserializer) {
+        Ok(value) => Ok(value),
+        Err(error) => {
+            relay_log::error!(
+                error = %error,
+                "Error deserializing global config option: {}",
+                std::any::type_name::<T>(),
```

Comment:
Is there a way to know the field name, to make the error provide more info? Example:
```
// Current
"Error deserializing global config option: bool
// Alternative
"Error deserializing global config option: span_usage_metric"
```

Reply:
Only by deserializing with [serde_path_to_error](https://docs.rs/serde_path_to_error) then the error would contain the key. The deserializer here is not aware of its context

Reply:
It should be good enough to not introduce another dependency, ship it!